packages/components/addon/components/hds/pagination-bar/index.js

```diff
@@ -1,14 +1,22 @@
 import Component from '@glimmer/component';
 import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
 
 export const TYPES = ['numbered', 'compact'];
 
 export default class HdsPaginationBarIndexComponent extends Component {
-  // UNCOMMENT THIS IF YOU NEED A CONSTRUCTOR
-  // constructor() {
-  //   super(...arguments);
-  //   // ADD YOUR ASSERTIONS HERE
-  // }
+  totalPages = Math.max(
+    Math.ceil(this.args.totalItems / this.args.itemsPerPage),
+    1
+  );
+  @tracked currentPage = 1;
+  @tracked itemsRangeStart =
+    (this.currentPage - 1) * this.args.itemsPerPage + 1;
```

Comment:
suggestion (based on experience): add as many comments as you can, when there is this kind of complex math/logic: in six months from now, I assure you, you will not remember why you did something in a certain way, what does it mean, what happens if you need to change something, etc.

make your future self happy 🙂  (and the people that will work on this code after you, too 😉)